src/js/memory/solverMemory.ts

```diff
@@ -0,0 +1,270 @@
+import { getById } from '../utilities';
+
+const position = "Position";
+const label = "Label";
+
+interface ElementData {
+	name: string;
+	stage: number;
+	value: number;
+  }
+
+export function memory_solve(e : HTMLInputElement): void {		
+	clean_form();
+
+	//only solve if checking a button
+	if (!e.checked) {
+		return;
+	}
+
+	var n = e.getAttribute("data-elementname");
+	var s =  e.getAttribute("data-elementstage");
+	var  v = e.getAttribute("data-elementvalue");
```

Comment:
Full words would be easier to read. Even if it's just `elName`, `elStage`, `elValue`.